### PR TITLE
Move unit test runner to standard library

### PR DESCRIPTION
**Pull request by presidento**
Move test runner to standard library.
Originated from #8819 

# After Submitting

I'll update the documentation about testing: http://www.nushell.sh/book/testing.html

### CODE REVIEW

**Comment by presidento** [comment]
@amtoine I put the run-tests command into `mod.nu`. If you want a separate file for it, please tell me how to do it.

File: crates/nu-std/lib/mod.nu [hunk 1]
```diff
@@ -155,3 +155,176 @@ export def clip [
...
+        msg: $"(ansi red)($error.msg)(ansi reset)"
+        label: {
+            text: ($error.label)
+            start: $error.span.start
+            end: $error.span.end
+        }
+    }
+}
+
+export def xxx [] {}
```
Comment (amtoine) [inline crates/nu-std/lib/mod.nu:199]: i think we do not want that :wink:

**Review by amtoine — CHANGES_REQUESTED** [review]
i have a look at how to put this in a module :+1:

**Comment by amtoine** [comment]
> @amtoine I put the run-tests command into `mod.nu`. If you want a separate file for it, please tell me how to do it.
>
something you can do
- let's say you move the commands from `crates/nu-std/lib/mod.nu` to something like `crates/nu-std/lib/testing.nu`
- then you can apply [`patch.diff`](https://github.com/nushell/nushell/files/11203214/patch.txt)

a bit hacky for now but that should work i think as `std testing run` :+1: 

> **Warning**
> you can NOT do `use std testing` and then `testing run` for now :relieved:

**Comment by presidento** [comment]
Applied, thanks. (I cannot test it in my current computer.)

File: crates/nu-std/lib/mod.nu (hunk 1 above)
Comment (presidento) [reply crates/nu-std/lib/mod.nu:200]: Yeah, sorry.

**Comment by presidento** [comment]
I moved the test-related stuff to `testing.nu`: the assert commands and the test runner

**Comment by presidento** [comment]
It failed with an unrelated error.

**Review by amtoine — APPROVED** [review]
hey, you know what, that looks good :star_struck: 

if we find things to tweak, we can address them later :relieved: 

thanks for all this and let's merge it :muscle:

**Pull request merged**